phpBB/phpbb/mention/source/friend.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license       GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\mention\source;
+
+class friend extends base_user
+{
+	/** @var  \phpbb\user */
+	protected $user;
+
+	/**
+	 * Set the user service used to retrieve current user ID
+	 *
+	 * @param \phpbb\user $user
+	 */
+	public function set_user(\phpbb\user $user)
+	{
+		$this->user = $user;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	protected function query($keyword, $topic_id)
+	{
+		/*
+		 * For optimization purposes all friends are returned regardless of the keyword
+		 * Names filtering is done on the frontend
+		 * Results will be cached on a per-user basis
+		 */
+		$query = $this->db->sql_build_query('SELECT', [
+			'SELECT'    => 'u.username_clean, u.username, u.user_id',
```

Comment:
you shouldn't return u.username as you don't use it

Reply:
It's used [here](https://github.com/phpbb/phpbb/pull/5225/files#diff-a9071ee38a42033caed608cc816e37e2R129).

Reply:
But I can get rid of this in queries and use the data from user_loader as it's being fetched anyway.
Fixed.